Composer/packages/client/src/store/persistence/Semaphor.ts

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+class Semaphore {
```

Comment:
So you used a Semaphaore to chain all async opertation right? 

what's the granularity of this? I mean, you make all actions synced, or just a set of actions in a certain transaction?

Reply:
The file persistence will not influence the actions. This is used to make a serial order for file operate. I have update it and now can concurrently run several task now.